test/libsolidity/Metadata.cpp

```diff
@@ -81,14 +80,13 @@ BOOST_AUTO_TEST_CASE(metadata_stamp_experimental)
 	BOOST_CHECK(dev::test::isValidMetadata(metadata));
 	bytes hash = dev::swarmHash(metadata).asBytes();
 	BOOST_REQUIRE(hash.size() == 32);
-	BOOST_REQUIRE(bytecode.size() >= 2);
-	size_t metadataCBORSize = (size_t(bytecode.end()[-2]) << 8) + size_t(bytecode.end()[-1]);
-	BOOST_REQUIRE(metadataCBORSize < bytecode.size() - 2);
+	bytes cborMetadata = dev::test::onlyMetadata(bytecode);
+	BOOST_REQUIRE(!cborMetadata.empty());
 	bytes expectation =
-		bytes{0xa2, 0x65, 'b', 'z', 'z', 'r', '0', 0x58, 0x20} +
+		bytes{0xa3, 0x65, 'b', 'z', 'z', 'r', '0', 0x58, 0x20} +
```

Comment:
In order to fix the tests, need to implement a helper: `parseCBORMetadata(bytes) -> map<string,string>`. This would parse strings into strings, bytes into hex strings and booleans into true/false strings.

The map could be then properly used in the tests.

Reply:
Done #6233.